factor/sms/classes/local/smsgateway/modica.php

```diff
@@ -0,0 +1,111 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Modica Group Gateway class
+ *
+ * @package     factor_sms
+ * @author      Alex Morris <alex.morris@catalyst.net.nz>
+ * @copyright   Catalyst IT
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace factor_sms\local\smsgateway;
+
+use admin_setting_configpasswordunmask;
+use admin_setting_configtext;
+
+defined('MOODLE_INTERNAL') || die();
+
+class modica implements gateway_interface {
+
+    /**
+     * Sends a message using the Modica Group Mobile Gateway API
+     *
+     * {@inheritDoc}
+     */
+    public function send_sms_message(string $messagecontent, string $phonenumber): bool {
+        global $CFG, $SITE;
+
+        $config = get_config('factor_sms');
+
+        $params = [
+            'http' => 'https://api.modicagroup.com/rest/gateway',
```

Comment:
maybe an admin setting? likely not needed just a thought

Reply:
I don't think it will change but I've added it anyway